docs/pages/api/rbac.mdx

```diff
@@ -0,0 +1,1017 @@
+```
+
+This should show that the Kubernetes Service is running:
+
+```text
+NAME               READY   STATUS    RESTARTS   AGE
+teleport-agent-0   1/1     Running   0          22s
+```
+
+`tsh` should indicate that the cluster has registered with Teleport:
+
+```code
+$ tsh kube ls
+Kube Cluster Name Labels           Selected
+----------------- ---------------- --------
+minikube          environment=demo
+```
+
+## Step 3/6. Write the client application
+
+At this point, we have set up an external RBAC system to use for generating
+Teleport roles and configured Teleport to allow our API client to interact with
+our Kubernetes cluster and Teleport cluster. In this step, we will write our
+client application.
+
+### Set up your Go project
+
+The first step is to create a project directory and initialize a Go module for
+your project: 
+
+```code
+$ mkdir sync-roles
+$ cd sync-roles
+$ go mod init sync-roles
+go: creating new go.mod: module sync-roles
+```
+
+This creates a new file in your project directory, `go.mod`, which Go's package
+management system uses to fetch your client application's dependencies.
+
+### Export identity files for the client application
+
+The `sync-kubernetes-rbac` user needs signed credentials in order to connect to
+your Teleport cluster as well as your Kubernetes cluster. You will use the `tctl
+auth sign` command to request these credentials for your API client.
+
+#### Connecting to your Teleport cluster
+
+The following `tctl auth sign` command impersonates the `sync-kubernetes-rbac`
+user, generates signed credentials, and writes an identity file to the local
+directory:
+
+```code
+$ tctl auth sign --user=sync-kubernetes-rbac --out=auth.pem
+```
+
+The identity file, `auth.pem`, includes both TLS and SSH credentials. Your
+client application uses the SSH credentials to connect to the Proxy Service,
+which establishes a reverse tunnel connection to the Auth Service. The client
+application uses this reverse tunnel, along with your TLS credentials, to
+connect to the Auth Service's gRPC endpoint.
+
+#### Connecting to the Kubernetes cluster
+
+You will also need to give the client application a way to authenticate to your
+Kubernetes cluster. To do this, use Teleport's certificate authority to sign
+credentials for the `sync-kubernetes-rbac` user. Your API client will present
+these credentials to authenticate to the Teleport Kubernetes Service, which will
+proxy requests to the Kubernetes cluster.
+
+Run the following command, ensuring that <Var name="proxy-address" /> includes
+the host and port of your Proxy Service:
+
+```code
+$ tctl auth sign --user=sync-kubernetes-rbac \
+  --kube-cluster-name=minikube \
+  --format=kubernetes \
+  --proxy=https://<Var name="proxy-address" /> \
+  --out=kubeconfig
+```
+
+### Add imports
+
+Create a file called `main.go` in your project directory with the following
+content:
+
+```go
+package main
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
```

Comment:
`ioutil` has been deprecated for a few major releases by now. Let's remove it (most of the functionality exists in `os` or `io`)